contracts/child/MaticChildERC20.sol

```diff
@@ -0,0 +1,95 @@
+pragma solidity ^0.5.2;
+
+import "./ERC20Mixin.sol";
+
+
+contract MaticChildERC20 is ERC20Mixin {
+
+  uint256 public currentSupply = 0;
+  uint8 constant private DECIMALS = 18;
+  bool isInitialized;
+
+  constructor() public {}
+
+  function initialize(address _childChain, address _token)
+    public {
+    // require(msg.sender == address(0x1000));
+    require(!isInitialized, "The contract is already initialized");
+    isInitialized = true;
+    token = _token;
+    _transferOwnership(_childChain);
+  }
+
+  function setParent(address ) public {
+    revert("Disabled feature");
+  }
+
+  function deposit(address user, uint256 amount) public onlyOwner {
+    // check for amount and user
+    require(amount > 0 && user != address(0x0), "Insufficient amount or invalid user");
+
+    // input balance
+    uint256 input1 = balanceOf(user);
+
+    // transfer amount to user
+    address payable _user = address(uint160(user));
+    _user.transfer(amount);
+
+    currentSupply = currentSupply.add(amount);
+
+    // deposit events
+    emit Deposit(token, user, amount, input1, balanceOf(user));
+  }
+
+  function withdraw(uint256 amount) public payable {
+    address user = msg.sender;
+    // input balance
+    uint256 input = balanceOf(user);
+
+    currentSupply = currentSupply.sub(amount);
+    // check for amount
+    require(amount > 0 && input >= amount && msg.value == amount, "Insufficient amount");
+
+    // withdraw event
+    emit Withdraw(token, user, amount, input, balanceOf(user));
+  }
+
+  function name() public view returns (string memory) {
+    return "Matic Token";
+  }
+
+  function symbol() public view returns (string memory) {
+    return "MATIC";
+  }
+
+  function decimals() public view returns (uint8) {
+    return DECIMALS;
+  }
+
+  function totalSupply() public view returns (uint256) {
+    return 10000000000 * 10 ** uint256(DECIMALS);
```

Comment:
Bro this is 100B. Commented yesterday too, where did that comment go 😆 Make it 10B.

Reply:
which school tell me this is 100B? 10,000,000,000 -> 1,000,000,000*10
added some comma for you why you are forcing me to count zeros lol did it 5-6 times now.